service/history/nDCStateRebuilder.go

```diff
@@ -123,27 +123,10 @@ func (r *nDCStateRebuilderImpl) rebuild(
 		return nil, 0, err
 	}
 
-	// need to specially handling the first batch, to initialize mutable state & state builder
```

Comment:
Was this dead code?  I don't think  I fully understand how this is related to re-applying signals.

Reply:
previously the version is required for mutable state initialization, now the dependency is removed, this change is to remove dead code